### PR TITLE
MGMT-9102: Remove unneeded test-infra release-ocm-2.4 jobs

### PR DESCRIPTION
**Pull request by eliorerz**
Save money by removing redundant jobs 

/cc @osherdp

### CODE REVIEW

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml [hunk 1]
```diff
@@ -65,18 +65,8 @@ tests:
...
-  steps:
-    cluster_profile: packet-assisted
-    env:
-      ASSISTED_CONFIG: |
-        SERVICE_BASE_REF=release-ocm-2.4
-        OPENSHIFT_VERSION="all"
-    workflow: baremetalds-assisted
 - as: e2e-metal-assisted-olm
   optional: true
-  run_if_changed: .*operator.*
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:79]: you'll need to remove this configuration from the job as well

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml [hunk 2]
```diff
@@ -100,7 +90,6 @@ tests:
     workflow: baremetalds-assisted
 - as: e2e-metal-assisted-ipv6
   optional: true
-  run_if_changed: ^(terraform_files/.*|.*/controllers/.*)$
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:103]: here too

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml [hunk 3]
```diff
@@ -109,20 +98,8 @@ tests:
...
-    cluster_profile: packet-assisted
-    env:
-      ASSISTED_CONFIG: |
-        SERVICE_BASE_REF=release-ocm-2.4
-        IPv6=yes
-        IPv4=yes
-    workflow: baremetalds-assisted
 - as: e2e-metal-assisted-none
   optional: true
-  run_if_changed: ^(terraform_files/.*|.*/controllers/.*)$
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:125]: here too

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml (hunk 1 above)
Comment (eliorerz) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:79]: It's not removed when running `make jobs`?

Comment (osherdp) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:79]: no

**Review by osherdp — APPROVED** [review]

**Pull request merged**